log4j-core/src/main/java/org/apache/logging/log4j/core/filter/RegexFilter.java

```diff
@@ -28,140 +25,312 @@
+     *   implementations, the message's formatted message will be returned.
+     * </p>
+     * <h3>Developer Note</h3>
+     * <p>
+     * While `Message#getFormat()` is broken in general, it still makes sense for certain types.
+     * Hence, suppress the deprecation warning.
+     * </p>
+     *
+     * @param message the message
+     * @return the target message based on configuration and message-type
+     */
+    @SuppressWarnings("deprecation")
+    private String getMessageTextByType(final Message message) {
+        return useRawMessage
+                        && (message instanceof ParameterizedMessage
+                                || message instanceof StringFormattedMessage
+                                || message instanceof StructuredDataMessage)
+                ? message.getFormat()
+                : message.getFormattedMessage();
     }
 
+    /** {@inheritDoc} */
     @Override
     public String toString() {
-        final StringBuilder sb = new StringBuilder();
-        sb.append("useRaw=").append(useRawMessage);
-        sb.append(", pattern=").append(pattern.toString());
-        return sb.toString();
+        return "useRawMessage=" + useRawMessage + ", pattern=" + pattern;
     }
 
     /**
-     * Creates a Filter that matches a regular expression.
-     *
-     * @param regex
-     *        The regular expression to match.
-     * @param patternFlags
-     *        An array of Strings where each String is a {@link Pattern#compile(String, int)} compilation flag.
-     * @param useRawMsg
-     *        If true, the raw message will be used, otherwise the formatted message will be used.
-     * @param onMatch
-     *        The action to perform when a match occurs.
-     * @param onMismatch
-     *        The action to perform when a mismatch occurs.
-     * @return The RegexFilter.
-     * @throws IllegalAccessException
-     * @throws IllegalArgumentException
+     * Creates a new builder instance.
+     * @return the new builder instance
      */
-    // TODO Consider refactoring to use AbstractFilter.AbstractFilterBuilder
     @PluginFactory
-    public static RegexFilter createFilter(
-            // @formatter:off
-            @PluginAttribute final String regex,
-            @PluginElement final String[] patternFlags,
-            @PluginAttribute final Boolean useRawMsg,
-            @PluginAttribute final Result onMatch,
-            @PluginAttribute final Result onMismatch)
-            // @formatter:on
-            throws IllegalArgumentException, IllegalAccessException {
-        if (regex == null) {
-            LOGGER.error("A regular expression must be provided for RegexFilter");
-            return null;
-        }
-        return new RegexFilter(useRawMsg, Pattern.compile(regex, toPatternFlags(patternFlags)), onMatch, onMismatch);
+    public static Builder newBuilder() {
+        return new Builder();
     }
```

Comment:
A plugin builder factory must be annotated with `@PluginBuilderFactory` instead of `@PluginFactory`.